src/mgr/ServiceMap.cc

```diff
@@ -75,20 +77,99 @@ std::string ServiceMap::Service::get_summary() const
   if (daemons.empty()) {
     return "no daemons active";
   }
+
   std::ostringstream ss;
-  ss << daemons.size() << (daemons.size() > 1 ? " daemons" : " daemon")
-     << " active";
 
-  if (!daemons.empty()) {
-    ss << " (";
+  // The format is one pair in metadata:
+  //   "classify" : "${type}/${prefix}"
+  //
+  // type: will be used to replace the default "daemon(s)"
+  // showed in `ceph -s`. If absent, the "daemon" will be used.
+  // prefix: if present the active members will be classified
+  // by the prefix instead of "daemon_name".
+  //
+  // For exmaple for iscsi gateways, it will be something likes:
+  //   "classify" + "\0" + "portal/gateway0" + "\0"
+  // The `ceph -s` will be something likes:
+  //    iscsi: 3 portals active (gateway0, gateway1, gateway2)
+
+  std::set<std::string> prefs;
+  bool classified = false;
+  auto p = daemons.begin();
+  auto q = p->second.metadata.find("classify");
```

Comment:
Nit: perhaps this could be a little more specifically named. Also, instead of the regex, why not just use two properties? 

i.e.: `{"daemon_type": "iscsi", "daemon_prefix": "portal1"}`

Reply:
Yeah, this looks better, will fix it.